runtime/lib/backends/nccl/device/distributed_backend_nccl.cc

```diff
@@ -154,26 +154,38 @@ Status DistributedBackendNCCL::all_to_all(const void *sendbuff, void *recvbuff,
 
 Status DistributedBackendNCCL::all_gather(const void *sendbuff, void *recvbuff,
                                           size_t sendlen, DTypeEnum dtype,
+                                          std::set<int64_t> replica_group,
                                           std::shared_ptr<DContext> ctx) {
   // check context type and get cuda stream
   assert(ctx->type() == "BRT_CTX_CUDA" && "only cuda context supported");
   cudaStream_t stream = static_cast<CudaContext *>(ctx.get())->get_stream();
+  int color = replica_group.find(rank()) == replica_group.end() ? 0 : 1;
```

Comment:
Please use NCCL_SPLIT_NOCOLOR to distinguish out the ranks not in the communication group. Ref: https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/usage/communicators.html